### PR TITLE
Fix: Anyscale working on FT

### PR DESCRIPTION
**Pull request by isaacbmiller**

### CODE REVIEW

File: dspy/clients/lm.py [hunk 1]
```diff
@@ -124,9 +123,12 @@ def kill(self):
         logger.info(msg)
 
     def finetune(self,
+            train_method: TrainingMethod,
             train_data: List[Dict[str, Any]],
+            eval_data: List[Dict[str, Any]] = None,
```
Comment (dilarasoylu) [inline dspy/clients/lm.py:128]: Extra?

File: dspy/clients/anyscale.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def wait_for_training(job_id):
     anyscale.job.wait(id=job_id, timeout_s=18000)
 
 
-def update_model_config(lora_dynamic_path: str, serve_config_path: str):
+def update_model_config(lora_dynamic_path: str, serve_config_path: str, job_id: str):
```
Comment (dilarasoylu) [inline dspy/clients/anyscale.py:163]: is `job_id` used?

File: dspy/clients/anyscale.py [hunk 3]
```diff
@@ -84,72 +81,75 @@ def status(self):
...
+        raise NotImplementedError(f"AnyScale can only support {TRAINING_METHODS_ANYSCALE} for the time being")
+
+    logger.info("[Finetune] Validating the dataset format...")
+    if not verify_dataset(train_data):
+        # TODO: Does AnyScale support text completion models?
+        err = "[Finetune] Error: Unable to verify that the dataset is in the correct format."
+        logger.error(err)
+        raise RuntimeError(err)
+
+    if eval_data and not verify_dataset(eval_data):
```
Comment (dilarasoylu) [inline dspy/clients/anyscale.py:106]: Delete?

File: dspy/clients/anyscale.py [hunk 4]
```diff
@@ -84,72 +81,75 @@ def status(self):
 def finetune_anyscale(
         job: FinetuneJobAnyScale,
         model: str,
+        method: TrainingMethod,
         train_data: List[Dict[str, Any]],
+        eval_data: List[Dict[str, Any]] = None,
```
Comment (dilarasoylu) [inline dspy/clients/anyscale.py:86]: Extra?

Comment (isaacbmiller) [reply dspy/clients/anyscale.py:86]: no? Maybe sometimes useful. Can also remove NBD

File: dspy/clients/finetune.py [hunk 5]
```diff
@@ -51,28 +51,33 @@ class TrainingStatus(str, Enum):
...
             model=self.model,
             train_data=self.train_data,
             train_kwargs=self.train_kwargs,
+            provider=self.provider,
+            method=self.method,
         )
 
     def __str__(self):
-        return f"FinetuningJob({self.to_dict()})"
+        return f"FinetuningJob({self.model})"
```
Comment (dilarasoylu) [inline dspy/clients/finetune.py:80]: The older version was useful to get more information about the job, but either works

Comment (isaacbmiller) [reply dspy/clients/finetune.py:80]: I was getting some issues with it, but I think they were red herrings - reverting

**Pull request merged**